### ISSUE TITLE
`hyrule.sequences` raises exception when `Sequence` is not imported.

### ISSUE THREAD

**Issue by Jlobblet** [hylang/hyrule#61]
When using various macros in `hyrule.sequences`, a `NameError` is raised complaining about missing name `Sequence`.

```hy
Hy 0.26.0 using CPython(main) 3.9.16 on Windows
=> (import hyrule)
=> hyrule.__version__
"0.3.0"
=> (require hyrule.sequences [seq])
=> (seq [n] (* n n))
Traceback (most recent call last):
  File "stdin-624a4dc7bdb62d48b1a0e703458e20f64431e1a0", line 1, in <module>
    (seq [n] (* n n))
NameError: name 'Sequence' is not defined
=> (import hyrule.sequences [Sequence])
=> (seq [n] (* n n))
[0, 1, 4, 9, 16, 25, 36, 49, 64, 81, 100, ...]
```

**Comment by Kodiologist** [comment]
It's the classic blunder of forgetting an `import` in the macro expansion. If you're interested in writing a pull request, I bet this would be easy.

**Comment by Jlobblet** [comment]
I'm new to Hy(rule) but happy to write a PR to address this. What is the best approach for `import`ing `Sequence`? Should it be bound with a symbol from `gensym` to avoid any potential collisions?
